lib/connectors_shared/constants.rb

```diff
@@ -11,4 +11,11 @@ class Constants
     ALLOW_FIELD = '_allow_permissions'.freeze
     DENY_FIELD = '_deny_permissions'.freeze
   end
+
+  class ConfigMetadata
```

Comment:
What's the goal of moving from the YAML to here?

Reply:
There was a problem with path resolution when I tried to do the bundle install for `ent-search` with the github link:
```
[!] There was an error while loading `connectors_sdk.gemspec`: No such file or directory - /Users/marynacherniavska/.rbenv/versions/jruby-9.3.3.0/lib/ruby/gems/shared/bundler/gems/connectors-04a3b5430910/lib/connectors_app/../../config/connectors.yml. Bundler cannot continue.

 #  from /Users/marynacherniavska/.rbenv/versions/jruby-9.3.3.0/lib/ruby/gems/shared/bundler/gems/connectors-04a3b5430910/connectors_sdk.gemspec:1
 #  -------------------------------------------
 >  require_relative 'lib/connectors_app/config'
 #
 #  -------------------------------------------
```
The problem was, `connectors.yml` isn't included in the gemspec for `connectors_sdk` gem, and also we should have had a default `connectors.yml` (not the `connectors.yml.example`) in the project. But instead, we (me and @artem-shelkovnikov) decided it would be better to break off the gemspec dependency on the config file.

Reply:
Indeed. It's also possible to move these 3 variables to a different yaml file but I would probably oppose the idea.

Conceptually what I felt was - our gemspec file is not compatible with one of the default ways to distribute gems (`gem git:` in gemfile), so we need to fix it. Fundamentally, the problem is as described - our build process is "custom" - dependent on a shell script that itself depends on some configuration files, and we cannot make it work with `gem git:` approach, it's limited.

It's also a question to us - do we actually need anything other than VERSION here?

Reply:
As long as these fields are in a single place in the whole project I think it's fine. But having the version in a static file makes our lives easier when we need to get that version (like how `ent-search` is doing) If you move it to a Ruby file, now you need Ruby to read it, it's harder to use from other places like bash scripts.

Reply:
I _think_ a gemspec is just a fancy ruby file - so arbitrary ruby code could be used. Could we just make parsing our gemspec actually do the same things as `make build` and then include the `connectors.yml` in the gemspec files?

> As long as these fields are in a single place in the whole project I think it's fine

I disagree - having this in code, and not configuration, and not updated by the build is not fine. It removes a solution key requirement of the Extractors As Dependencies project, which was to be able to clearly tell if a customer had forked our repo, and to help the customer know (from a support perspective) which version of their fork they are running on. Relying on the customer to make certain code changes is too brittle.

Reply:
>  Relying on the customer to make certain code changes is too brittle.

But if they fork and change the code, I think we can make the assumption they have the developer hat on and can change the version wherever we put that version

Reply:
Moving these changes back into `yml` due to the considerations above.